diff -Naur zato/code/zato-server/src/zato/server/connection/amqp/outgoing.py work/code/zato-server/src/zato/server/connection/amqp/outgoing.py
--- zato/code/zato-server/src/zato/server/connection/amqp/outgoing.py	2017-07-06 06:16:36.297979268 +0000
+++ work/code/zato-server/src/zato/server/connection/amqp/outgoing.py	2017-07-10 06:46:25.297979268 +0000
@@ -206,7 +206,7 @@
             properties[name] = value
 
         if not properties['expiration']:
-            properties['expiration'] = 0
+            properties.pop('expiration', None)
         else:
             properties['expiration'] = int(properties['expiration'])
 
